consensus/messages.go

```diff
@@ -4,35 +4,34 @@ import (
 	"log"
 
 	typesCons "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/shared/codec"
 	"github.com/pokt-network/pocket/shared/crypto"
-	"google.golang.org/protobuf/proto"
 )
 
 func CreateProposeMessage(
-	m *ConsensusModule,
-	step typesCons.HotstuffStep, // step can be taken from `m` but is specified explicitly via interface to avoid ambiguity
+	height uint64,
+	round uint64,
+	step typesCons.HotstuffStep,
+	block *typesCons.Block,
 	qc *typesCons.QuorumCertificate,
 ) (*typesCons.HotstuffMessage, error) {
-	if m.Block == nil {
-		return nil, typesCons.ErrNilBlockProposal
-	}
-
 	msg := &typesCons.HotstuffMessage{
```

Comment:
shouldn't we still guard for `block == nil` like before?

```go
	if block == nil {
		return nil, typesCons.ErrNilBlockProposal
	}
```

Reply:
Yessir. Ty!